datalad/distribution/tests/test_add.py

```diff
@@ -1,477 +0,0 @@
-    # without a base dataset the next is interpreted as "add everything
-    # in subds to subds"
-    add(subds.path)
-    ok_clean_git(subds.path)
-    assert_not_in('dir', ds.subdatasets(result_xfm='relpaths'))
-    # but with a base directory we add the dataset subds as a subdataset
-    # to ds
-    ds.add(subds.path)
-    assert_in('dir', ds.subdatasets(result_xfm='relpaths'))
-    #  create another one
-    other = create(other)
-    # install into superdataset, but don't add
-    other_clone = install(source=other.path, path=opj(ds.path, 'other'))
-    ok_(other_clone.is_installed)
-    assert_not_in('other', ds.subdatasets(result_xfm='relpaths'))
-    # now add, it should pick up the source URL
-    ds.add('other')
-    # and that is why, we can reobtain it from origin
-    ds.uninstall('other')
-    ok_(other_clone.is_installed)
-    ds.get('other')
-    ok_(other_clone.is_installed)
-
-
-@known_failure_githubci_win
-@with_tree(tree={
-    'file.txt': 'some text',
-    'empty': '',
-    'file2.txt': 'some text to go to annex',
-    '.gitattributes': '* annex.largefiles=(not(mimetype=text/*))'}
-)
-def test_add_mimetypes(path):
-    # XXX apparently there is symlinks dereferencing going on while deducing repo
-    #    type there!!!! so can't use following invocation  -- TODO separately
-    import os
-    path = os.path.realpath(path)  # yoh gives up for now
-    ds = Dataset(path).create(force=True)
-    ds.repo.add('.gitattributes')
-    ds.repo.commit('added attributes to git explicitly')
-    # now test that those files will go into git/annex correspondingly
-    __not_tested__ = ds.add(['file.txt', 'empty'])
-    ok_clean_git(path, untracked=['file2.txt'])
-    # Empty one considered to be  application/octet-stream  i.e. non-text
-    ok_file_under_git(path, 'empty', annexed=True)
-    ok_file_under_git(path, 'file.txt', annexed=False)
-
-    # But we should be able to force adding file to annex when desired
-    ds.add('file2.txt', to_git=False)
-    ok_file_under_git(path, 'file2.txt', annexed=True)
-
-
-@with_tempfile(mkdir=True)
-def test_gh1597_simpler(path):
-    ds = Dataset(path).create()
-    # same goes for .gitattributes
-    with open(opj(ds.path, '.gitignore'), 'a') as f:
-        f.write('*.swp\n')
-    ds.add('.gitignore')
-    ok_clean_git(ds.path)
-    ok_file_under_git(ds.path, '.gitignore', annexed=False)
-    # put .gitattributes in some subdir and add all, should also go into Git
-    os.makedirs(op.join(ds.path, 'subdir'))
-    attrfile = op.join(ds.path, 'subdir', '.gitattributes')
-    with open(attrfile, 'a') as f:
-        f.write('# just a comment\n')
-    ds.add('.')
-    ok_clean_git(ds.path)
-    ok_file_under_git(ds.path, op.relpath(attrfile, start=ds.path), annexed=False)
-
-
-# Failed to run ['git', '--work-tree=.', 'diff', '--raw', '-z', '--ignore-submodules=none', '--abbrev=40', 'HEAD', '--'] This operation must be run in a work tree
-@with_tempfile(mkdir=True)
-def test_gh1597(path):
```

Comment:
I think at least some tests need just to migrate into using new save to guarantee that we don't breed regressions etc

Reply:
All relevant tests have been ported to the new implementation last year already.